src/CommunityToolkit.Aspire.Hosting.Zitadel/ZitadelResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting;
+using Aspire.Hosting.ApplicationModel;
+
+namespace CommunityToolkit.Aspire.Hosting.Zitadel;
+
+/// <summary>
+/// Provides extension methods for adding Zitadel resources to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class ZitadelResourceBuilderExtensions
+{
+    private const string AdminEnvVarName = "ZITADEL_DEFAULTINSTANCE_ORG_HUMAN_USERNAME";
+    private const string AdminEnvVarPassword = "ZITADEL_DEFAULTINSTANCE_ORG_HUMAN_PASSWORD";
+
+    private const int DefaultcontainerPort = 8080; // ZITADEL_PORT
+    private const string ManagementEndpointName = "management";
+
+    /// <summary>
+    /// Adds a Zitadel container to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. </param>
+    /// <param name="port">The host port that the underlying container is bound to when running locally.</param>
+    /// <param name="adminUsername">The parameter used as the admin for the Zitadel resource. If <see langword="null"/> a default value will be used.</param>
+    /// <param name="adminPassword">The parameter used as the admin password for the Zitadel resource. If <see langword="null"/> a default password will be used.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<ZitadelResource> AddZitadel
+    (
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int? port = null,
+        IResourceBuilder<ParameterResource>? adminUsername = null,
+        IResourceBuilder<ParameterResource>? adminPassword = null
+    )
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        var passwordParameter = adminPassword?.Resource ??
+                                builder.AddParameter
+                                (
+                                    name: $"{name}-password",
+                                    valueGetter: () => "Password1!",
```

Comment:
Take a look at how the other resources implement default passwords. None are hardcoded and we do random password generation. https://github.com/dotnet/aspire/blob/cf14299e30838ccfd369ee416c5d962409fd4e5a/src/Aspire.Hosting.MySql/MySqlBuilderExtensions.cs#L35

Reply:
Saw that. Wasn't sure if I should use the documented default value or generate it. I'll switch to the generated password.